CRM/Contribute/BAO/ContributionPage.php

```diff
@@ -389,8 +389,13 @@ public static function sendMail($contactID, $values, $isTest = FALSE, $returnMes
         $tplParams['onBehalfName'] = $displayName;
         $tplParams['onBehalfEmail'] = $email;
 
+        $template->assign('onBehalfProfile', NULL);
         if (!empty($values['onbehalf_profile_id'])) {
-          self::buildCustomDisplay($values['onbehalf_profile_id'], 'onBehalfProfile', $contactID, $template, $params['onbehalf_profile'], $fieldTypes);
+          [$groupTitle, $values] = self::getProfileNameAndFields($values['onbehalf_profile_id'], $contactID, $params['onbehalf_profile'], $fieldTypes);
```

Comment:
$values is already used elsewhere in this function and this will overwrite it.
Suggest $onbehalfProfileFields or something like that.